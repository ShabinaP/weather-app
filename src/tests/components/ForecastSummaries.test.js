import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 3333444,
      description: "stub description 1",
      icon: "stubIcon1",
      temperature: {
        max: 22,
        min: 14,
      },
    },
    {
      date: 3344245,
      description: "stub description 2",
      icon: "stubicon3",
      temperature: {
        max: 35,
        min: 10,
      },
    },
  ];
  it("renders correctly", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={validProps} />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
