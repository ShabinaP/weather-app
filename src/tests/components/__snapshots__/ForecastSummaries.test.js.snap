// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        3333444
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        stubIcon1
      </div>
      <div
        class="forecast-summary__temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary__description"
      >
        stub description 1
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        3344245
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        stubicon3
      </div>
      <div
        class="forecast-summary__temperature"
      >
        35°C
      </div>
      <div
        class="forecast-summary__description"
      >
        stub description 2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
